cellpose/core.py

```diff
@@ -77,7 +77,7 @@ def assign_device(use_torch=True, gpu=False, device=0):
     if isinstance(device, str):
         if device != "mps" or not(gpu and torch.backends.mps.is_available()):
             device = int(device)
-    if gpu and use_gpu(use_torch=True):
+    if gpu and use_gpu(gpu_number=device, use_torch=True):
```

Comment:
IMO this should be use_gpu(gpu_number=device,use_torch=use_torch) to propagate all params passed to assign_device. If you don't care about use_torch why even have it as a parameter.